demos/unity_states/unity/Assets/Scripts/BallStateDataSourcePlatform.cs

```diff
@@ -0,0 +1,202 @@
+using UnityEngine;
+using TP = Teleportal.Client;
+using States = Teleportal.Client.Contract.Properties.States;
+using RSharp;
+
+public class BallStateDataSourcePlatform : MonoBehaviour, IBallStateDataSource
+{
+    // Platform interop types
+    private TP.Baseline baselineMain;
+    private BallContract ballContract;
+    private TP.Object.ObjectHandle ballObject;
+
+    private States.StateHandle_F32 stateHandlePosX;
+    private States.StateHandle_F32 stateHandlePosY;
+    private States.StateHandle_F32 stateHandlePosZ;
+    private States.StateHandle_I16 stateHandleEulerX;
+    private States.StateHandle_I16 stateHandleEulerY;
+    private States.StateHandle_I16 stateHandleEulerZ;
+    private States.StateHandle_F32 stateHandleScaleX;
+    private States.StateHandle_F32 stateHandleScaleY;
+    private States.StateHandle_F32 stateHandleScaleZ;
+    private States.StateHandle_U64 stateHandleColor;
+
+    void Awake()
+    {
+        InstantiatePlatformObject();
```

Comment:
style nit: 

My opinions (which stem in part from watching many years ago that video I posted the other day in slack lol) are that its actually preferable to have long functions for "procedural" code, instead of pulling them out into other helper methods just to visually break up or group commands. If you want to group the code, use comment sections, `{}` blocks, or even nested functions or closures *inside* the larger function.

The rationale for this is that pulling code out into a separate function that isn't actually going to be reused anywhere else just makes it harder to directly see the control flow, and you have to jump around in your IDE a lot. It also increases the burden of naming that helper function - in this PR I had nits on the naming of those already 😅

Feel free to disregard or discuss further, but this is my personal style opinion after too often seeing a billion trivial helper methods that make code harder rather than eaiser to read. 
Also, if anyone ever quotes "uncle" bob martin, [run and don't listen to them](https://youtu.be/7EmboKQH8lM?list=PLmmYSbUCWJ4x1GO839azG_BBw8rkh-zOj&t=3442)

Reply:
I've been thinking about this and while I understand the rationale for putting everything of a procedure in one function, I don't necessarily agree in most situations. For functions that would contain 2 or 3 lines of code, I agree that It's just overhead. When there are tens of lines of code doing something, however, I think it still makes sense to split.

Will merge `InstantiatePlatformObject()` and `ConfigurePlatformObjectStates()` on the premise that their contents are both part of instantiating the Platform object, and the configuring of states is only done once.

`Awake()` is part of the Unity MonoBehaviour lifecycle (which includes Start, Update, and many others) so in my mind it doesn't make sense to clutter those lifecycle functions. It's all just style though :) Lmk what you think of the compromise mentioned in the previous paragraph.

Reply:
The other nice thing about naming functions is that the code becomes more self-documenting (although the same could be achieved with a comment I suppose)

Reply:
> the same could be achieved with a comment
![image](https://user-images.githubusercontent.com/6969415/182238982-c9a99572-9618-4318-968f-4a33abcd336a.png)


pros: 
* same local scoping of variables as functions due to the `{..}`
* Discrete groups/steps of work are grouped, just like in a function
* don't need to assign to fields of class to pass data to a subsequent function (`this.ballObject`)
* block still foldable/collapsible
* **all control flow is local and no jumping around in IDE**

cons: a bit more indentation. Can always break it into a function if it becomes bad.

Reply:
as for `Awake` being part of `MonoBehavior` lifecycle, I didn't understand what you meant - your function is still doing this logic one way or another, regardless of if it calls out into another function. Perhaps I am overlooking something?

If you still feel like you don't want to adopt this style thats totally ok! I won't be totalitarian in this, and feel free to mark as resolved. But hopefully what I was suggesting is more clear and its food for thought :)

Reply:
Makes sense :) Updating.